modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
-				return null;
-			}
-		}
-		//receive messages
-		try {
 
-			GetResponse delivery = null;
-			delivery = channel.basicGet(queueName, false);
-
-			if (delivery != null) {
-				//deserilizing message
-				StorableMessage storableMessage = null;
-				ByteArrayInputStream bis = new ByteArrayInputStream(delivery.getBody());
-				ObjectInput in = new ObjectInputStream(bis);
-				try {
-					storableMessage = (StorableMessage) in.readObject();
-				} catch (ClassNotFoundException e) {
-					logger.error(getId() + "unable to read the stored message" + e);
-					channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
-				}
+    private static final Log log = LogFactory.getLog(RabbitMQConsumer.class.getName());
+
+    private Connection connection;
+    private Channel channel;
+    private RabbitMQStore store;
+    private String queueName;
+    private String idString;
+    private CachedMessage cachedMessage; // Holds the last message read from the message store
+
+    public RabbitMQConsumer(RabbitMQStore store) {
+        if (store == null) {
+            log.error("Cannot initialize.");
+            return;
+        }
+        this.store = store;
+        cachedMessage = new CachedMessage();
+    }
 
-				bis.close();
-				in.close();
-				org.apache.axis2.context.MessageContext axis2Mc = store.newAxis2Mc();
-				MessageContext synapseMc = store.newSynapseMc(axis2Mc);
-				synapseMc =
-						MessageConverter.toMessageContext(storableMessage, axis2Mc, synapseMc);
-				updateCache(delivery, synapseMc, null, false);
-				if (logger.isDebugEnabled()) {
-					logger.debug(getId() + " Received MessageId:" +
-					             delivery.getProps().getMessageId());
-				}
-				return synapseMc;
-			}
-		} catch (ShutdownSignalException sse) {
-			logger.error(getId() + " connection error when receiving messages" + sse);
-		} catch (IOException ioe) {
-			logger.error(getId() + " connection error when receiving messages" + ioe);
-		}
-		return null;
-	}
+    /**
+     * Get a single message from the queue and deserialize for set into the message context
+     *
+     * @return the {@link MessageContext} with received message
+     */
+    @Override
+    public MessageContext receive() {
+        if (isAlive()) {
+            try {
+                GetResponse delivery = channel.basicGet(queueName, false);
+                if (delivery != null) {
+                    StorableMessage storableMessage = deserializeMessage(channel, delivery);
+                    org.apache.axis2.context.MessageContext axis2Mc = store.newAxis2Mc();
+                    MessageContext synapseMc = store.newSynapseMc(axis2Mc);
+                    synapseMc = MessageConverter.toMessageContext(storableMessage, axis2Mc, synapseMc);
+                    updateCache(delivery, delivery.getProps().getMessageId());
+                    log.info(getId() + " Received MessageId: " + delivery.getProps().getMessageId());
```

Comment:
Do we need this log, which we'll get per message?

Reply:
Ack. I added it for testing purpose and forgot to remove.